### PR TITLE
Error prompt when i upload a wrong file format for media files #330

### LINKED ISSUE #330: 🎨 ❌  As a user I should get an error prompt when I upload a wrong file format for media files
## User stories

When we upload a wrong file to either the image, video or document block we don't get a message saying we uploaded the wrong kind of file.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Summary

When we upload a wrong file to either the image, video or document block we don't get a message saying we uploaded the wrong kind of file. 

### Description

For this, we will need a dialog box that is displayed when a user uploaded an unsupported media file. 

The modal should be created under `libs/elements/layout/modals`

### PR DESCRIPTION
**Pull request by 23po**
# Description

Created a reusable dialog box under libs/elements/layout/modals that is displayed when a user uploaded an unsupported media file.

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #330

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# Video Link
[error modal edit.webm](https://user-images.githubusercontent.com/48943229/224250056-766e3101-5a2e-4c09-8594-b8a23e962690.webm)

# How Has This Been Tested?

This runs on the development server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by 23po** [comment]
@ReaganCn @LemmyMwaura @IanOdhiambo9 please review

File: libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts [hunk 1]
```diff
@@ -8,8 +8,13 @@ import { StoryBlockTypes } from '@app/model/convs-mgr/stories/blocks/main';
 
 import { UploadFileService } from '@app/state/file';
 
+import { MatDialog } from '@angular/material/dialog';
+
+
 import { _JsPlumbComponentDecorator } from '@app/features/convs-mgr/stories/blocks/library/block-options';
 
+import { ErrorPromptModalComponent } from 'libs/elements/layout/modals/src/lib/modals/error-prompt-modal/error-prompt-modal.component';
+
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts:17]: Import from the aliased path

File: libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts [hunk 2]
```diff
@@ -42,8 +49,46 @@ export class DocumentBlockComponent implements OnInit {
...
+    if (!event.target.files || !event.target.files[0]) {
+      this.docLink = this.defaultLink;
+      return;
+    }
+
+    if (event.target.files[0].type !== 'application/pdf') {
+      // Show error modal
+      alert('Please select a PDF file.');
+      return;
+    }
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts:88]: Use the error Modal

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/components/video-block/video-block.component.ts [hunk 3]
```diff
@@ -11,6 +11,10 @@ import { UploadFileService } from '@app/state/file';
 
 import { _JsPlumbComponentDecorator } from '@app/features/convs-mgr/stories/blocks/library/block-options';
 
+import { MatDialog } from '@angular/material/dialog';
+
+import { ErrorPromptModalComponent } from 'libs/elements/layout/modals/src/lib/modals/error-prompt-modal/error-prompt-modal.component';
+
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/components/video-block/video-block.component.ts:17]: Import from aliased path

File: libs/features/convs-mgr/stories/blocks/library/image-message-block/src/lib/components/image-block/image-block.component.ts [hunk 4]
```diff
@@ -11,6 +11,10 @@ import { UploadFileService } from '@app/state/file';
 
 import { _JsPlumbComponentDecorator } from '@app/features/convs-mgr/stories/blocks/library/block-options';
 
+import { MatDialog } from '@angular/material/dialog';
+
+import { ErrorPromptModalComponent } from 'libs/elements/layout/modals/src/lib/modals/error-prompt-modal/error-prompt-modal.component';
+
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/image-message-block/src/lib/components/image-block/image-block.component.ts:17]: Also here fix Import path.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Just some minor fixes

File: libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts (hunk 1 above)
Comment (23po) [reply libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts:17]: fixed

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/components/video-block/video-block.component.ts (hunk 3 above)
Comment (23po) [reply libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/components/video-block/video-block.component.ts:17]: fixed

File: libs/features/convs-mgr/stories/blocks/library/image-message-block/src/lib/components/image-block/image-block.component.ts (hunk 4 above)
Comment (23po) [reply libs/features/convs-mgr/stories/blocks/library/image-message-block/src/lib/components/image-block/image-block.component.ts:17]: fixed

File: libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts (hunk 2 above)
Comment (23po) [reply libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts]: fixed

**Comment by 23po** [comment]
@LemmyMwaura issues have been fixed

File: libs/features/convs-mgr/stories/blocks/library/image-message-block/src/lib/components/image-block/image-block.component.ts [hunk 5]
```diff
@@ -39,11 +43,13 @@ export class ImageBlockComponent implements OnInit {
...
-              public domSanitizer: DomSanitizer
-  ) {
-    this.block = this.block as ImageMessageBlock;
-  }
+
+  constructor(private _imageUploadService: UploadFileService,
+    public domSanitizer: DomSanitizer,
+    private dialog: MatDialog
+) {
+this.block = this.block as ImageMessageBlock;
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/image-message-block/src/lib/components/image-block/image-block.component.ts:51]: ??

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/convs-mgr-video-message-block.module.ts [hunk 6]
```diff
@@ -13,6 +15,8 @@ import {
 
 import {MultiLangModule}from '@ngfi/multi-lang';
 
+import { ElementsLayoutModalsModule } from '@app/elements/layout/modals';
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/convs-mgr-video-message-block.module.ts:18]: No need to import this module. Applies to the video message module also

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
- There's an issue with the feature. I can only check the file type once. After that, the modal doesn't show up. 
- Also if I encounter an error during upload the spinner never stops. 
- The open modal method since it's essentially the same in every file can be saved in a shared service and then we pass the dynamic values, such as title. Please use the FileStorage service.

**Pull request closed without merging**